src/main/java/com/google/codeu/data/Message.java

```diff
@@ -30,15 +32,18 @@
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
    * random ID and uses the current system time for the creation time.
    */
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  //add recipient to the constructor
+  public Message(String user, String text, String recipient) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
   }
 
-  public Message(UUID id, String user, String text, long timestamp) {
+  //add recipient to the constructor
+  public Message(UUID id, String user, String text, long timestamp, String recipient) {
```

Comment:
Looks like you win here, and have to add new javadoc for this constructor.

Intellij should actually do the right thing if you type /** <tab>  in terms of adding the @param values of each of the arguments.  You'll still have to add high level documentation describing what this constructor is supposed to do.

Reply:
Got it!